application/uitests/source/class/uitests/DragAndDropApplication.js

```diff
@@ -0,0 +1,156 @@
+/* ************************************************************************
+
+   Copyright:
+
+   License:
+
+   Authors:
+
+ ************************************************************************ */
+
+/**
+ * This is the main application class of your custom application "dragndrop"
+ * 
+ * @asset(qx/icon/Tango/16/actions/document-print.png)
+ */
+qx.Class.define("uitests.DragAndDropApplication", {
+  extend: qx.application.Standalone,
+
+  members: {
+    main: function() {
+      this.base(arguments);
+
+      if (qx.core.Environment.get("qx.debug")) {
+        qx.log.appender.Native;
+        qx.log.appender.Console;
+      }
+
+      var doc = this.getRoot();
+      var tb = new qx.ui.toolbar.ToolBar();
+      var btn = new qx.ui.toolbar.Button("Click Me", "qx/icon/Tango/16/actions/document-print.png");
+      tb.add(btn);
+      doc.add(tb, { left: 10, top: 10, right: 10 });
+      
+      var root = new qx.ui.container.Composite(new qx.ui.layout.HBox(10));
+      doc.add(root, { left: 10, top: 100, right: 10, bottom: 10 });
+
+      var compSource = new qx.ui.container.Composite(new qx.ui.layout.VBox(6));
+      root.add(compSource);
+      compSource.add(new qx.ui.basic.Label("Simple Source"));
+      var lstSource = new qx.ui.form.List().set({ draggable: true, selectionMode: "multi" });
+      compSource.add(lstSource);
+
+      for (var i = 0; i < 20; i++)
+        lstSource.add(new qx.ui.form.ListItem("Item " + i, "icon/16/places/folder.png"));
+
+      var cbxCancelDragstart = new qx.ui.form.CheckBox("Cancel dragstart").set({ value: false });
+      compSource.add(cbxCancelDragstart);
+      
+      var cbxRejectDragstart = new qx.ui.form.CheckBox("Reject dragstart").set({ value: false });
+      compSource.add(cbxRejectDragstart);
+      
+      lstSource.addListener("dragstart", function(e) {
+        this.debug("lstSource: dragstart");
+        return new qx.Promise((resolve, reject) => {
+          if (cbxCancelDragstart.getValue())
+            e.preventDefault();
+          if (cbxRejectDragstart.getValue())
+            return reject();
```

Comment:
`return` with a value here; without a value later. `reject()` doesn't actually return anything. This should change to:
```
reject();
return;
```